includes/tools/array_parser.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Customisation Database package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+use battye\array_parser\parser;
+
+class array_parser
+{
+	/**
+	 * Find the array of subscribed events
+	 * @param $file_path
+	 * @return array
+	 * @throws Exception
+	 */
+	public static function check_events($file_path)
+	{
+		// Find the events, if there are any
+		$regex = '/\bclass\s+\S+\s+implements\s+EventSubscriberInterface\s*{\s*(?:[^{}]*{[^{}]*})*[^{}]*\bgetSubscribedEvents\(\)\s*{[^{}]*?(?:\barray\s*\(|\[)([^{}]*?)(?:\)|\])[^{}]*}/x';
+		$result = parser::parse_regex($regex, $file_path);
```

Comment:
*That* is super interesting and probably something you worked on for a long time.
Does this also work for other accepted return values?
I mean, not just `'event' => 'function'` array entries but also:
```php
'event' => array(
    array('function', int priority),
    array('function', int priority)
)
```
?

Reply:
I think it should be fine in that format. All we need is the event name (which is the key), so the format of the value won't matter. It ends up being called like this:

```php	
$matches = \array_parser::check_events($file->getRealPath());
$event_list = array_keys($matches);
```